datasource/src/main/scala/quasar/physical/mongo/expression/Core.scala

```diff
@@ -16,78 +16,60 @@
 
 package quasar.physical.mongo.expression
 
-import slamdata.Predef.{String => SString, Int => SInt, _}
-
-import matryoshka.Delay
-
-import monocle.Prism
+import slamdata.Predef.{Eq => _, _}
 
 import quasar.{RenderTree, NonTerminal, Terminal}
 
-import scalaz.{Traverse, Applicative, Scalaz}, Scalaz._
-import scalaz.syntax._
+import cats._
+import cats.implicits._
+import higherkindness.droste.Delay
+import higherkindness.droste.util.DefaultTraverse
+import monocle.Prism
 
-trait Core[A] extends Product with Serializable
+sealed trait Core[A] extends Product with Serializable
 
 object Core {
-  final case class Array[A](arr: List[A]) extends Core[A]
-  final case class Object[A](obj: Map[SString, A]) extends Core[A]
-  final case class Bool[A](bool: Boolean) extends Core[A]
-  final case class Int[A](int: SInt) extends Core[A]
-  final case class String[A](string: SString) extends Core[A]
+  final case class Array[A](value: List[A]) extends Core[A]
+  final case class Object[A](value: Map[String, A]) extends Core[A]
+  final case class Bool[A](value: Boolean) extends Core[A]
+  final case class SInt[A](value: Int) extends Core[A]
+  final case class SString[A](value: String) extends Core[A]
```

Comment:
Not sure if you like how this pans out, but whenever there are duplicate names like this I'm trying to import like `import slamdata.{Predef => s}` so you can define as 
```
final case class Int[A](value: s.Int) extends Core[A]
final case class String[A](value: s.String) extends Core[A]
```

Maybe just me liking this construction, feel free to leave as-is.

Reply:
:+1: